src/Bicep.Core/Analyzers/Linter/Rules/InterpolateNotConcatRule.cs

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Parsing;
+using Bicep.Core.PrettyPrint;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class InterpolateNotConcatRule : LinterRuleBase
+    {
+        public new const string Code = "prefer-interpolation";
+
+        public InterpolateNotConcatRule() : base(
+            code: Code,
+            description: CoreResources.InterpolateNotConcatRuleDescription,
+            docUri: "https://aka.ms/bicep/linter/prefer-interpolation")
+        { }
+
+        private CodeReplacement GetCodeReplacement(TextSpan span)
+            => new CodeReplacement(span, $"{Code} - this is the new code");
+
+        override internal IEnumerable<IBicepAnalyzerDiagnostic> AnalyzeInternal(SemanticModel model)
+        {
+            var visitor = new Visitor(this);
+            visitor.Visit(model.SyntaxTree.ProgramSyntax);
+            return visitor.diagnostics;
+        }
+
+        private class Visitor : SyntaxVisitor
+        {
+            public List<IBicepAnalyzerDiagnostic> diagnostics = new List<IBicepAnalyzerDiagnostic>();
+
+            private InterpolateNotConcatRule parent;
+
+            public Visitor(InterpolateNotConcatRule parent)
+            {
+                this.parent = parent;
+            }
+
+            public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
+            {
+                if (syntax.NameEquals("concat"))
```

Comment:
>NameEquals [](http://example.com/codeflow?start=27&length=10)

This will break when we have user-defined functions or another namespace has `concat`.

Reply:
I would expect NameEquals to eventually take an optional argument for namespace, and if no namespace is passed in, it would assume namespace is empty/default.  I.e., this should still work at that point, no?  There's currently no concept of other namespaces, right?

Reply:
We do have namespaces already. Most functions are in the `sys` namespace with a bunch more in `az`. They are automatically imported in every compilation. The fully qualified version of `concat()` is `sys.concat()`. In fact, the syntax node is an `InstanceFunctionCallSyntax` then, so there might be a bug with that also.

A match on function name alone is sufficient currently because we don't have overlaps, but results in a new class of bugs once more namespaces are available and clashes happen. I want to avoid that entirely. That said, I'm not sure we have a convenient helper that does this for you right now, so we'd need to introduce one. It should look up the function by name in the specified namespace and then check that the syntax node is bound to the same symbol via `ReferenceEquals`.

Reply:
The imported namespaces are available in the `FileSymbol` instance.

Reply:
Can be handled as a bug.

Reply:
https://github.com/Azure/bicep/issues/2773